### PR TITLE
feat: improve `Map::get()` @throws inference

### PR DESCRIPTION
**Pull request by simPod**
Opening based on https://github.com/phpstan/phpstan/issues/9066, hope it's correct place where to put the extension

### CODE REVIEW

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Please provide a test that fails without the extension (and also test the code that is supposed to throw the exception).

File: tests/PHPStan/Analyser/data/throw-points/ext-ds.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	}
+};
+
+function () {
+	$map = new \Ds\Map();
+	try {
+		$map->get('1', null);
+		$foo = 1;
+	} finally {
+		assertVariableCertainty(TrinaryLogic::createYes(), $foo);
```
Comment (simPod) [inline tests/PHPStan/Analyser/data/throw-points/ext-ds.php]: Any idea why this failed? I've debugged the extension and it returns VoidType in this case so phpstan should know nothing is thrown 🤔

Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/data/throw-points/ext-ds.php]: No idea. You can try to show the effect of the extension in a rule test (like a dead catch...).

File: tests/PHPStan/Rules/Exceptions/data/ext-ds.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace PHPStan\Rules\Exceptions\data;
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Exceptions/data/ext-ds.php]: Bad namespace and filename, please use `namespace Bug9066` + `bug-9066.php`.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**